### PR TITLE
added `tf:optional` annotation to simplify the code

### PR DESCRIPTION
**Pull request by alexott**
Quite often there are situations when field should be required in the JSON payload (so we send default values, such as `true`, `0`, ...) but it's really optional in the Terraform schema, so we need to explicitly modify the schema with code like:

```
m["enable_photon"].Optional = true
m["enable_photon"].Required = false
```

This patch adds the `optional` keyword that will be checked in the `tf` annotation, so we can avoid explicit schema modification

### CODE REVIEW

**Review by nfx — COMMENTED** [review]
this is a great change, though it requires a bit thorough integration testing. let's aim it for 1.1, as it changes schemas slightly

**Comment by alexott** [comment]
it's covered by the unit test...

**Comment by alexott** [comment]
@nfx I've added another test for SQL Endpoint that demonstrates that `tf:optional` works

**Comment by nfx** [comment]
image

**Review by nfx — CHANGES_REQUESTED** [review]
Can you add few more unit tests?

File: clusters/clusters_api.go [hunk 1]
```diff
@@ -207,7 +207,7 @@ type InitScriptStorageInfo struct {
 	Dbfs *DbfsStorageInfo `json:"dbfs,omitempty" tf:"group:storage"`
 	Gcs  *GcsStorageInfo  `json:"gcs,omitempty" tf:"group:storage"`
 	S3   *S3StorageInfo   `json:"s3,omitempty" tf:"group:storage"`
-	File *LocalFileInfo   `json:"file,omitempty" tf:"optional"`
+	File *LocalFileInfo   `json:"file,omitempty"`
```
[the remark this answers is not in the document]
Comment (alexott) [reply clusters/clusters_api.go:210]: yes, schema is the same

**Review by nfx — APPROVED** [review]

**Pull request merged**